substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/SecurityServicesFeature.java

```diff
@@ -678,10 +686,10 @@ private static Function<String, Class<?>> getConstructorParameterClassAccessor(I
                 if (engineDescription == null) {
                     return null;
                 }
-                String constrParamClassName = (String) consParamClassNameField.get(engineDescription);
-                if (constrParamClassName != null) {
-                    return loader.findClass(constrParamClassName).get();
+                if (consParamClassFieldFinal.getName().equals("constructorParameterClassName")) {
+                    return loader.findClass((String) consParamClassFieldFinal.get(engineDescription)).get();
```

Comment:
We are missing a null-check here. I.e. `consParamClassFieldFinal.get(engineDescription)` can be `null`.

I wonder why we've done this custom fix at all instead of just downporting and adapting [svm: adopt "JDK-8324646: Avoid Class.forName in SecureRandom constructor](https://github.com/oracle/graal/commit/05c1e79d360a9ab375eac31d1ce6946a311d74bc).

I've created a corresponding [PR for graalvm-for-jdk21-community-backports](https://github.com/graalvm/graalvm-for-jdk21-community-backports/pull/4). Maybe we can take that change instead of this one?

Reply:
@simonis OK. Thanks. Please create a PR fixing this. We've merged this change already so would prefer a follow-up fix.

> I wonder why we've done this custom fix at all instead of just downporting and adapting [svm: adopt "JDK-8324646: Avoid Class.forName in SecureRandom constructor](https://github.com/oracle/graal/commit/05c1e79d360a9ab375eac31d1ce6946a311d74bc).

It would be a custom change anyway as the version conditionals would need to change. We usually go with a version agnostic approach (e.g. testing if the `constructor.ParameterClassName` field is there. If the original fix would have been done that way it would have been a no-change backport. Those `if <VERSION_1> do_thing1() else do_thing2()` are a bit problematic as it needs to change with every JDK update that a JDK fix might get backported.

Reply:
I agree that a version agnostic approach is preferable. Has this been discussed with the GraalVM upstream project already?

I'll create a PR to fix this in Mandrel now.

Reply:
> I agree that a version agnostic approach is preferable. Has this been discussed with the GraalVM upstream project already?

It was originally an ask from them when contributing patches. Not sure why they've back-tracked on that.

> I'll create a PR to fix this in Mandrel now.

Thanks!